lib/rubocop/cop/rails/root_pathname_methods.rb

```diff
@@ -214,8 +214,17 @@ def include_interpolation?(arguments)
         end
 
         def join_arguments(arguments)
-          quote = include_interpolation?(arguments) ? '"' : "'"
-          joined_arguments = arguments.map(&:value).join('/')
+          use_interpolation = false
+
+          joined_arguments = arguments.map do |arg|
+            if arg.respond_to?(:value)
+              arg.value
+            else
+              use_interpolation = true
+              "\#{#{arg.source}}"
+            end
+          end.join('/')
+          quote = include_interpolation?(arguments) || use_interpolation ? '"' : "'"
```

Comment:
I wonder if we should bother to have all this extra logic to encode the default style of `Style/StringLiterals` in this cop, or just pass responsibility to that cop for people who have it enabled?

Reply:
Good point. OTOH, it follows the existing behavior and doesn't seem to have a complicated logic, so I think this cop can cover it. Anyway I will open a separate PR for `Style/StringLiterals` enforced style as it is different from the issue reported in #870.